src/components/Attachment/components/WaveProgressBar.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { MouseEventHandler, useMemo, useRef, useState } from 'react';
+import clsx from 'clsx';
+import { divMode } from '../utils';
+
+type WaveProgressBarProps = {
+  /** Function that allows to change the track progress */
+  seek: MouseEventHandler<HTMLDivElement>;
+  /** The array of fractional number values between 0 and 1 representing the height of amplitudes */
+  waveformData: number[];
+  /** Allows to specify the number of bars into which the original waveformData array should be resampled */
+  amplitudesCount?: number;
+  /** Progress expressed in fractional number value btw 0 and 100. */
+  progress?: number;
+};
+export const WaveProgressBar = ({
+  amplitudesCount = 40,
+  progress = 0,
+  seek,
+  waveformData,
+}: WaveProgressBarProps) => {
+  const [progressIndicator, setProgressIndicator] = useState<HTMLDivElement | null>(null);
+  const isDragging = useRef(false);
+
+  const handleMouseDown = () => {
+    if (!progressIndicator) return;
+    isDragging.current = true;
+    progressIndicator.style.cursor = 'grabbing';
+  };
+
+  const handleMouseMove: MouseEventHandler<HTMLDivElement> = (event) => {
+    if (!isDragging.current) return;
+    seek(event);
+  };
+
+  const handleMouseUp = () => {
+    if (!progressIndicator) return;
+    isDragging.current = false;
+    progressIndicator.style.removeProperty('cursor');
+  };
+
+  const resampledWaveformData = useMemo(
+    () =>
+      waveformData.length === amplitudesCount
+        ? waveformData
+        : waveformData.length > amplitudesCount
+        ? downSample(waveformData, amplitudesCount)
+        : upSample(waveformData, amplitudesCount),
+    [amplitudesCount, waveformData],
+  );
+
+  if (!waveformData.length) return null;
+
+  return (
+    <div
+      className='str-chat__wave-progress-bar__track'
+      data-testid='wave-progress-bar-track'
+      onClick={seek}
+      onMouseDown={handleMouseDown}
+      onMouseLeave={handleMouseUp}
+      onMouseMove={handleMouseMove}
+      onMouseUp={handleMouseUp}
+      role='progressbar'
+    >
+      {resampledWaveformData.map((amplitude, i) => (
+        <div
+          className={clsx('str-chat__wave-progress-bar__amplitude-bar', {
+            ['str-chat__wave-progress-bar__amplitude-bar--active']:
+              progress > (i / resampledWaveformData.length) * 100,
+          })}
+          data-testid='amplitude-bar'
+          key={`amplitude-${i}`}
+          style={
+            {
+              '--str-cha__wave-progress-bar__amplitude-bar-height': amplitude
```

Comment:
```suggestion
              '--str-chat__wave-progress-bar__amplitude-bar-height': amplitude
```
I think this is a typo :)

Reply:
Yes, resolving in CSS and in the SDK.